hpx/runtime/serialization/array.hpp

```diff
@@ -65,8 +65,8 @@ namespace hpx { namespace serialization
         template <class Archive>
         void serialize(Archive& ar, unsigned int v)
         {
-            typedef std::integral_constant<bool,
-                hpx::traits::is_bitwise_serializable<T>::value> use_optimized;
+            typedef typename
+                hpx::traits::is_bitwise_serializable<T>::type use_optimized;
```

Comment:
This (and other similar changes) should just be `hpx::traits::is_bitwise_serializable<T>`, as the trait would be required to derive from `std::integral_constant<bool, C>` as other traits do.